### DISCUSSION TITLE
RFC round-up 7/22/2020: across step, task queue, worker pools

### DISCUSSION THREAD

**Discussion by vito** [concourse/concourse#5909] [Ideas]
Hi all, I'm going to try something different for this one: posting it as a Discussion instead!

Previous RFC round-ups can be found on [our blog](https://blog.concourse-ci.org/tag/rfcs/). To be honest our blog doesn't get a whole lot of natural traffic at the moment, so I'm hoping this provides better visibility and lowers the barrier to commenting and asking questions and such.

## RFCs in need of feedback

If you have any thoughts on these RFCs, please contribute them as review comments on the RFC PR! :pray:

### [RFC #29: `across` step](https://github.com/concourse/rfcs/pull/29)

This RFC is pretty much good to go; there are no open questions. Feedback welcome!

The `across` step is a key component to [v10](https://blog.concourse-ci.org/core-roadmap-towards-v10/), providing the mechanism for setting pipelines 'across' all pull requests, branches, and even matrix-style var combinations.

  
### [RFC #43: task queue](https://github.com/concourse/rfcs/pull/43)

The task queue RFC proposes a solution to the longstanding issue of Concourse yeeting a bunch of workloads onto workers at once, overloading them and causing builds to fail.

Open questions:

* Is "Resource Pool" a good name? (resource means something different in the context of pipeline jobs and tasks)
* Could the functionality of the "Resource Pool"' be fit some already existing component?
* Should this be tasks scheduling or more generic step scheduling? That is to include check containers (resource-type checks etc.)
* What is the relationship to the new LIDAR component?

### [RFC #62: worker pools](https://github.com/concourse/rfcs/pull/62)

Right now, workers can be registered either cluster-wide or for a single team. This RFC introduces a 'worker pool' - a group of workers which can be associated to multiple teams. This will prove especially important if/when we decide to [merge Projects and Teams](https://github.com/vito/rfcs/blob/5679ee86b4e810a8ae8e00dcb21aca2ad745103e/032-projects/proposal.md#can-projects-replace-teams).

Open questions:

* Do we really need `--public-worker-key` option for `fly set-worker-pool`?

## Upcoming RFCs

* I'm working on an RFC for webhooks! There's a *lot* of nuance to suss out here, but I hope to have something published for the next blog post. I'll be talking about it in [#dev](https://discord.gg/VR76HG3)!
* As mentioned in the last RFC round-up I'll be submitting an RFC soon for the `run` step, originally introduced in concourse/rfcs#37 but it likely needs more detail (that RFC mainly focused on the Prototype protocol).
* I have a rough idea for improving the UX for TV CI radiators, README badges, and things like CCmenu. I touched on this in my replies to #5308 - if you're interested in chatting about that, I'll be in [#dev](https://discord.gg/VR76HG3)!

---

If you've submitted an RFC and would like me to include it in the next round-up, let me know! Feel free to link to it in the replies to this Discussion.

Thanks!

**Comment by evanchaoli** [comment]
@vito https://github.com/concourse/rfcs/pull/66 please.
